### PR TITLE
Fix factor edge cases in check_values() (#298 review)

### PR DESCRIPTION
**Pull request by joethorley**
Addresses review comments on #298 and subsequent revisions agreed in review. Changes on top of the `issue-297` branch.

## Behaviour

`check_values()` (and hence `check_data()`) now applies the following rules for a factor `values`:

| `values` | Meaning |
|---|---|
| `factor()` | any factor (class check only) |
| `factor(1)` | any factor without missing values |
| `factor(c(1, NA))` | any factor with or without missing values |
| `factor(NA)` | any factor with all missing values |
| two or more levels | levels of x must be identical, including order, to the levels of values (with missing values governed by the presence of `NA` in values) |

This keeps the stricter level checking from #297/#298 for two or more levels while restoring the established one-level idioms (`factor(1)`, `factor(c(1, NA))`), which a reverse-dependency check showed `klexdatr` relies on.

## Fixes to #298

- `check_values(factor("a"), factor())` passes again: an empty factor checks class only, as documented.
- Removed `chk_gt(nlevels(x))`, which leaked the internal variable name in its error message (```nlevels(x)` must be greater than 0, not 0.``) and failed zero-length columns.
- Removed `chk_subset(unique(values), c(levels(values), NA))`, which was dead code: a factor's values are by construction its levels or `NA`.

## Other changes

- Dropped "(the first occurence of)" from the `chk_orderset()` error message (breaking change to the message text), so orderset failures now read: ```levels(data$fac)` must have each of the following elements in the following order: 'a', 'b', 'z'.`` No reverse dependency matches the old text (checked locally and via GitHub code search).
- Updated the `check_values()` `@details` to document the factor rules and the three one-level idioms.
- Added a `@details` cross-reference from `check_data()` to `check_values()` for the value-checking rules.
- Added direct `check_values()` tests for the fixed edge cases and revised behaviours (all written test-first).

## Verification

- Full suite: 1282 tests pass with no warnings.
- Reverse-dependency check: data-check test suites of `bisonpictools`, `klexdatr`, `kootlake`, `juvenilegerrard`, `obsbiascomp` and `ssddata` all pass against this branch (the one `bisonpictools` test error is pre-existing and unrelated).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: man/check_values.Rd [hunk 1]
```diff
@@ -36,9 +36,13 @@ To check the range of the values in x pass two non-missing values
 To check that x only includes specific values
 pass three or more non-missing values.
 
-In the case of a factor ensure values has two levels to
-check that the levels of x are an ordered superset of the levels of value
-and three or more levels to check that they are identical.
+In the case of a factor if values has two or more levels
```
Comment (StefanoMezzini) [inline man/check_values.Rd]:
```suggestion
In the case of a factor if \code{values} has two or more levels
```

File: man/check_values.Rd [hunk 2]
```diff
@@ -36,9 +36,13 @@ To check the range of the values in x pass two non-missing values
...
+then the levels of x must be identical,
```
Comment (StefanoMezzini) [inline man/check_values.Rd]:
```suggestion
then the levels of \code{x} must be identical,
```

File: man/check_values.Rd [hunk 3]
```diff
@@ -36,9 +36,13 @@ To check the range of the values in x pass two non-missing values
...
+including their order, to the levels of values.
+A factor with less than two levels only checks that x is a factor.
```
Comment (StefanoMezzini) [inline man/check_values.Rd]:
```suggestion
A factor with less than two levels only checks that \code{x} is a factor.
```

File: R/check-values.R [hunk 4]
```diff
@@ -20,9 +20,13 @@
 #' To check that x only includes specific values
 #' pass three or more non-missing values.
 #'
-#' In the case of a factor ensure values has two levels to
-#' check that the levels of x are an ordered superset of the levels of value
-#' and three or more levels to check that they are identical.
+#' In the case of a factor if values has two or more levels
```
Comment (StefanoMezzini) [inline R/check-values.R]:
```suggestion
#' In the case of a factor if `values` has two or more levels
```

File: R/check-values.R [hunk 5]
```diff
@@ -20,9 +20,13 @@
...
+#' then the levels of x must be identical,
```
Comment (StefanoMezzini) [inline R/check-values.R]:
```suggestion
#' then the levels of `x` must be identical,
```

File: R/check-values.R [hunk 6]
```diff
@@ -20,9 +20,13 @@
...
+#' including their order, to the levels of values.
```
Comment (StefanoMezzini) [inline R/check-values.R]:
```suggestion
#' including their order, to the levels of `values`.
```

File: man/check_values.Rd [hunk 7]
```diff
@@ -36,9 +36,13 @@ To check the range of the values in x pass two non-missing values
 To check that x only includes specific values
 pass three or more non-missing values.
 
-In the case of a factor ensure values has two levels to
-check that the levels of x are an ordered superset of the levels of value
-and three or more levels to check that they are identical.
+In the case of a factor if values has two or more levels
+then the levels of x must be identical,
+including their order, to the levels of values.
```
Comment (StefanoMezzini) [inline man/check_values.Rd]:
```suggestion
including their order, to the levels of \code{values}.
```

File: R/check-values.R [hunk 8]
```diff
@@ -20,9 +20,13 @@
...
+#' A factor with less than two levels only checks that x is a factor.
```
Comment (StefanoMezzini) [inline R/check-values.R]:
```suggestion
#' A factor with less than two levels only checks that `x` is a factor.
```

File: tests/testthat/test-check-data.R
Comment (StefanoMezzini) [inline tests/testthat/test-check-data.R]:
```suggestion
# level doesn't matter: only checking that class is "factor" when values is empty factor
expect_no_error(check_data(data.frame(fac = factor(c("a", "b", "c", "z"))),
                           list(fac = factor())))

# level doesn't matter: only checking that class is "factor" when values is 1 level and not NA
expect_no_error(check_data(data.frame(fac = factor(c("a", "b", "c", "z"))),
                           list(fac = factor("z"))))
```

File: tests/testthat/test-check-values.R [hunk 9]
```diff
@@ -42,6 +42,22 @@ test_that("check_values pass", {
...
+    check_values(factor(c(1, NA)), factor(c(2, NA)))
+  )
+  expect_identical(
+    check_values(factor(NA), factor(NA)),
+    check_values(factor(NA), factor(NA))
+  )
+  expect_identical(
+    check_values(factor(character(0)), factor("z")),
+    check_values(factor(character(0)), factor("z"))
+  )
```
Comment (StefanoMezzini) [inline tests/testthat/test-check-values.R]:
```suggestion
```

**Review by StefanoMezzini — CHANGES_REQUESTED** [review]

**Comment by StefanoMezzini** [comment]
the updated behaviour for handling factors makes sense.

**Comment by joethorley** [comment]
I accepted all changes

**Review by StefanoMezzini — APPROVED** [review]
